demo/openssl-qat-engine/Dockerfile

```
@@ -1,4 +1,30 @@
-FROM ubuntu:24.04
+FROM ubuntu:24.04 AS builder
+
+ARG QAT_ENGINE_VERSION="v1.6.2"
 
 RUN apt update && \
-    apt install --no-install-recommends -y qatengine qatlib-examples qatzip openssl
+    env DEBIAN_FRONTEND=noninteractive apt install -y \
+    git \
+    autoconf \
+    g++ \
+    pkg-config \
+    libssl-dev \
+    libtool \
+    make \
```

I'd prefer to add `libqat-dev` explicit too since it's needed also:
```suggestion
    make \
    libqat-dev \
```